document-warehouse/search-document.js

```diff
@@ -0,0 +1,82 @@
+/** Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+async function main(
+  projectNumber = 'YOUR_PROJECT_NUMBER',
+  location = 'YOUR_PROJECT_LOCATION',
+  userId = 'user:xxx@example.com',
+  documentQueryText = 'YOUR_DOCUMENT_QUERY'
+) {
+  // [START contentwarehouse_search_documents]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   * const projectNumber = 'YOUR_PROJECT_NUMBER';
+   * const location = 'YOUR_PROJECT_LOCATION'; // Format is 'us' or 'eu'
+   * const userId = 'user:xxx@example.com'; // Format is "user:xxx@example.com"
+   * const documentQueryText = 'YOUR_DOCUMENT_QUERY'
+   */
+
+  // Import from google cloud
+  const {DocumentServiceClient} = require('@google-cloud/contentwarehouse').v1;
+
+  // Create service client
+  const serviceClient = new DocumentServiceClient();
+
+  // Get Document Schema
+  async function quickstart() {
+    // Initialize request argument(s)
+    const searchRequest = {};
+
+    // The full resource name of the location, e.g.:
+    // projects/{project_number}/locations/{location}
+    searchRequest.parent = `projects/${projectNumber}/locations/${location}`;
+
+    // File Type Filter
+    const fileTypeFilter = {};
+    fileTypeFilter.fileType = 'DOCUMENT';
+
+    // Document Text Query
+    const documentQuery = {};
+    documentQuery.query = documentQueryText;
+    documentQuery.fileTypeFilter = fileTypeFilter;
+    searchRequest.documentQuery = documentQuery;
+
+    // Histogram Query
+    const histogramQuery = {};
+    histogramQuery.histogramQuery = 'count("DocumentSchemaId")';
+    searchRequest.histogramQueries = [histogramQuery];
+
+    // Metadata Definition
+    searchRequest.requestMetadata = {userInfo: {id: userId}};
```

Comment:
```suggestion
    const searchRequest = {
      // The full resource name of the location, e.g.:
      // projects/{project_number}/locations/{location}
      parent: `projects/${projectNumber}/locations/${location}`,
      // Document Text Query
      documentQuery: {
        query: documentQueryText,
        // File Type Filter
        fileTypeFilter: {
          fileType: 'DOCUMENT'
        }
      },
      // Histogram Query
      histogramQueries: [
        {
          histogramQuery: 'count("DocumentSchemaId")'
        }
      ],
      requestMetadata: {userInfo: {id: userId}}
    };
```

This setup is preferred as it makes the object structure easier to follow.

Note - I couldn't find a method for creating the project/location string, so I left it as is.

https://cloud.google.com/nodejs/docs/reference/contentwarehouse/latest/contentwarehouse/v1.documentserviceclient